lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -987,20 +962,35 @@ func (d *zfs) SetVolumeQuota(vol Volume, size string, allowUnsafeResize bool, op
 	}
 
 	// Handle filesystem datasets.
-	key := "quota"
-	if shared.IsTrue(vol.ExpandedConfig("zfs.use_refquota")) {
-		key = "refquota"
+	for _, property := range []string{"quota", "refquota", "reservation", "refreservation"} {
+		err = d.setDatasetProperties(d.dataset(vol, false), fmt.Sprintf("%s=none", property))
+		if err != nil {
+			return err
+		}
 	}
 
 	value := fmt.Sprintf("%d", sizeBytes)
 	if sizeBytes == 0 {
-		value = "none"
+		return nil
+	}
+
+	quota_key := "quota"
+	reservation_key := "reservation"
```

Comment:
Should be `quotaKey` and `reservationKey` as Go doesn't recommend snake case.

Reply:
fixed